tests/Integration/inc/Engine/Cache/AdminSubscriber/addWpCacheStatusTest.php

```diff
@@ -16,20 +16,29 @@ class Test_AddWpCacheStatusTest extends AdminTestCase {
 	/**
 	 * @dataProvider providerTestData
 	 */
-	public function testShouldAddWpCacheTest( $tests, $expected ) {
+	public function testShouldAddWpCacheTest( $config, $tests, $expected ) {
         $result = apply_filters( 'site_status_tests', $tests );
 
-        $this->assertArrayHasKey(
-            'wp_cache_status',
-            $result['direct']
-        );
-
-        $this->assertSame(
-            $expected['direct']['wp_cache_status']['label'],
-            $result['direct']['wp_cache_status']['label']
-        );
-
-        $this->assertTrue( is_callable( $result['direct']['wp_cache_status']['test'] ) );
+        if ( isset( $config['filter_constant_value'] ) ) {
+			add_filter( 'rocket_set_wp_cache_constant', [ $this, 'return_false' ] );
+		}
+
+        if ( isset( $expected['direct'] ) ){
+			$this->assertArrayHasKey(
+                'wp_cache_status',
+                $result['direct']
+            );
+    
+            $this->assertSame(
+                $expected['direct']['wp_cache_status']['label'],
+                $result['direct']['wp_cache_status']['label']
+            );
+    
+            $this->assertTrue( is_callable( $result['direct']['wp_cache_status']['test'] ) );
+		}
+		else{
+			$this->assertTrue( is_array( $result ) );
```

Comment:
```suggestion
			$this->assertArrayNotHasKey( 'wp_cache_status', $result['direct'] );
```

That way we check specifically for the value we don't want to be added.

Reply:
@Tabrisrp I'm not sure why Integration test is failing.